examples/search.c

```diff
@@ -0,0 +1,196 @@
+			NULL, &nih);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlGetId(nih, &id);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc(nih, PTL_PT_FLOWCTRL, eqh, PTL_PT_ANY, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	/* describe the memory used to receive the message */
+	me = (ptl_me_t){ .start = NULL,
+			.length = 0,
+			.ct_handle = PTL_CT_NONE,
+			.uid = PTL_UID_ANY,
+			.options = PTL_ME_OP_PUT,
+			.match_id = PTL_NID_ANY,
+			.match_bits = 0,
+			.ignore_bits = 0,
+			.min_free = 0 };
+	
+	/* expose to the network the reception queue */
+	ret = PtlMEAppend(nih, pti, &me, PTL_OVERFLOW_LIST, NULL, &meh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto me_unlink;
+	} else {
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s\n", msg);
+	}
+
+	/* describe the memory used to send the message */
+	md = (ptl_md_t){ .start = NULL,
+			.length = 0,
+			.options = 0,
+			.eq_handle = eqh,
+			.ct_handle = PTL_CT_NONE };
+
+	ptl_hdr_data_t header_data = 0;
+
+	ret = PtlMDBind(nih, &md, &mdh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlMDBind failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto me_unlink;
+	}
+
+	/* send the message to ourself */
+	ret = PtlPut(mdh, 0, 0, PTL_ACK_REQ, id, pti, 0, 0, NULL, header_data);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPut failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+
+	printf("The message has been sent !\n");
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s\n", msg);
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s\n", msg);
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto md_release;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s\n", msg);
+
+	printf("Let's search the message in the reception queue...\n");
+
+	/* try to find me */
+	ret = PtlMESearch(nih, pti, &me, PTL_SEARCH_ONLY, NULL);

```

Comment:
Let's perform two searches:
- One with invalid match bits, that ought to fail
- One with the correct match bits, that ought to succeed